### PR TITLE
Hulks no longer oneshot literally every clockwork structure

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscdel: Hulks no longer one-shot most clockwork structures and will no longer four-shot the Gateway to the Celestial Derelict.
bugfix: Hulks are no longer a blob counter.
/:cl:

Hulk damage is now an object proc!
Mecha now use that proc!

### CODE REVIEW

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	var/force = 0
 
 	var/list/armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+	var/hulk_damage = 150 //the damage hulks do on punches to this object, is affected by melee armor
```
Comment (phil235) [inline code/game/objects/objs.dm]: making a var for every single obj for a problem that's very specific is bad. Use a proc instead.

Comment (ChangelingRain) [inline code/game/objects/objs.dm]: I think using a proc for this is absurd, and really it should be an armor value.

**Comment by Niknakflak** [comment]
Meh no :-1:

**Comment by Durkel** [comment]
The real hero would be the guy who replaces hulk with a new power so we don't have to deal with spaghetti hulk all over the code.

**Comment by optimumtact** [comment]
just reduce the damage a hulk is doing

**Comment by phil235** [comment]
The whole problem is consistency, we want hulk able to break walls and airlocks easily, but somehow clockwork structure should be hard to smash as hulk.

File: code/game/objects/objs.dm (hunk 1 above)
Comment (GunHog) [inline code/game/objects/objs.dm]: I am speaking from ignorance here, but could you not override `attack_hulk()` for steampunk structures for reduced damage?

**Comment by optimumtact** [comment]
well then maybe joan should just accept this as legit counterplay and add some anti hulk capability to the clockwork spells list.

**Comment by ChangelingRain** [comment]
hulks are absurd and I already did with guvax, because fuck hulks stuns are NOT ENOUGH
and no, fuck hulks, hulks don't get counterplay, they get nerfs.

**Comment by optimumtact** [comment]
this isn't a nerf, it's a snowflake hack

just give all your structures like 100 percent melee resist armour or something

File: code/game/objects/objs.dm (hunk 1 above)
Comment (phil235) [inline code/game/objects/objs.dm]: e.g. proc/get_hulk_damage() return 42

**Comment by ChangelingRain** [comment]
I don't want my structures to super resist melee, though!

**Comment by optimumtact** [comment]
THEN EXPECT THE HULK TO FUCK THEM UP

**Comment by AdamElTablawy** [comment]
**HNNNNGH!!**

**Comment by PKPenguin321** [comment]
>hulks are absurd and I already did with guvax, because fuck hulks stuns are NOT ENOUGH
and no, fuck hulks, hulks don't get counterplay, they get nerfs.

hahahahaha git needs a salt emoji

File: code/game/objects/objs.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/objects/objs.dm]: this is an idiotic solution

**Comment by ChangelingRain** [comment]
all of these possible solutions to hulks are ridiculous it's either a var, which phil hates, a proc, which I think is just fucking absurd, or ~hulk armor values for hulk type damage~ which holy shit i'd need to go through all armor, everywhere

**Comment by Militaires** [comment]
how about

and listen to this, it may seem absurd at first cause it's a new concept to you but listen

how about

you dont actually nerf every possible counterplay to clock cult and accept that there will be gaps in the design like everything in the game does instead of powercreeping it to resist every possible tactic or method to combat

how about that.

**Comment by ChangelingRain** [comment]
Well, it turns out being able to destroy clockcult's big gateway objective in four goddamn hits is bad for game balance, so fuck off.

**Comment by Militaires** [comment]
@ChangelingRain 
the key to handling hulk as clock cult (if you're letting a hulk get next to your gateway in the first place)

is to git gud.

okay maybe 4 hits for giant portal is bad but for machines it's not that bad.

**Comment by ChangelingRain** [comment]
Everything uses phil damage, so everything takes 150 modified by melee armor, which is, uh, difficult to balance anything around.
Thus this pr.

**Comment by Militaires** [comment]
@ChangelingRain 

can you just make it a proc to check if attacker is hulk and lower damage instead of adding ANOTHER damage var to keep track off

it may be dirty but it works better

**Comment by ChangelingRain** [comment]
that's what this pr is doing

**Comment by Militaires** [comment]
@ChangelingRain 

much better good job

**Comment by ExcessiveUseOfCobblestone** [comment]
without trying to turn this pr into a hulk discussion, most of the time those who have hulk also have xray,spacewalk capabilities, tk, stealth, and dwarfism [aka a smaller sprite to click on] so regardless a validhunting geneticist is going to be a pain even if they're not in melee distance of clock areas.

also hulk is dumb, replace with J E L L Y B O N E S

**Comment by ChangelingRain** [comment]
genetics is dumb and i'm trying as hard as I can to mitigate that

**Pull request merged**